### ISSUE TITLE
Try-catch errors with misspelling in SQL

### ISSUE THREAD

**Issue by addlistener** [AlaSQL/alasql#1526]
@mathiasrw Check it out. You can fork it to see the error messages.

The server breaks when there's no such function. Ideally it should be caught by the try block.

https://replit.com/@jchnxu/DeficientSubstantialSystemsanalysis#index.js

**Comment by jchnxu** [comment]
<img width="888" alt="image" src="https://user-images.githubusercontent.com/4531670/197472354-54299ace-58b8-4c98-b8fc-f249ef517c54.png">
maybe we should add some try catch here?

**Comment by jchnxu** [comment]
I used a proxy now to hack it. But there will be concurrency issues later.
```
        // TODO: concurrency issue here when multiple requests?
        alasql.fn = new Proxy(originalAlasqlFn, {
          get(target, name, receiver) {
            try {
              if (Reflect.has(target, name)) {
                return Reflect.get(target, name, receiver);
              } else {
                throw new Error(
                  "Function " + name.toString() + " not implemented."
                );
              }
              return "missing";
            } catch (e) {
              reject(e);
            }
          },
        });
```

**Comment by mathiasrw** [comment]
Hmm. I cant replicate this. Any chance you can make a jsfiddle with it? https://jsfiddle.net/8brvex4f/ 

I dont understand where use `median` - and what is `wrongFuncName(email_domain)`?


---
**Bonus info**
Alasql is async and returns a promise if you call `alasql.async` or if you put the first argument in an array, or if you engage with a file like CSV or excel. So instead of the function in the second parameter you can do something like 

```
app.use(async ctx => {
  ctx.body = 'Hello World';
  const { columns, data: objects } = await alasql.promise("SELECT RECORDSET wrongFuncName(`email_domain`) FROM CSV('some.csv') AS `temp`")
  });
});
```

**Comment by mathiasrw** [comment]
Might be related to https://github.com/AlaSQL/alasql/issues/1491 - try v2.1.5 and see if that helps

**Comment by mathiasrw** [comment]
Ahhhhhh

**Comment by Chaitanyaprasad60** [comment]
Hi, Seems like issue is not yet resolved. Can you assign it to me?

**Comment by Chaitanyaprasad60** [comment]
I found a way to catch the error and due to this, the server will not go down. But I am unable to return the error through the callback function. As "cb" is not available in the scope of the function. Any ideas ??

**Comment by mathiasrw** [comment]
What function does not have the callback avaliable?

**Comment by Chaitanyaprasad60** [comment]
yy.Select.prototype.compileSelect2 This Function. Where the error is actually occurring

**Comment by mathiasrw** [comment]
Fix released as part of v4.1.5
